core/counting/shardedCounts.go

```diff
@@ -0,0 +1,49 @@
+package counting
+
+import (
+	"fmt"
+	"strings"
+)
+
+var _ Counts = (*ShardedCounts)(nil)
+
+// ShardedCounts keeps counts for a sharded data structure
+type ShardedCounts struct {
+	total   int64
+	byShard map[string]int64
+}
+
+// NewShardedCounts creates a new ShardedCounts
+func NewShardedCounts() *ShardedCounts {
+	return &ShardedCounts{
+		byShard: make(map[string]int64),
+	}
+}
+
+// PutCounts registers counts for a shard
+func (counts *ShardedCounts) PutCounts(shardName string, value int64) {
+	counts.byShard[shardName] = value
```

Comment:
Is it not possible to access this and/or GetTotal concurrently? 
otherwise you would need a mutex when accessing the map

later edit: I see you use these under backingMap mutex, then please add a comment that this implementation is not concurrent safe

Reply:
Indeed, the counts object is not concurrent safe, it is supposed to be populated the its owner (the one who constructs it) and then (afterwards) just read from whatever routine. I will make it concurrent safe.